### DISCUSSION TITLE
Docker setup, trying to deploy to two Synology NASes and one SSH server

### DISCUSSION THREAD

**Discussion by macdis** [acmesh-official/acme.sh#5010] [Q&A] [answered]
I use acme.sh in a Docker container on Synology NAS no. 1, not as a daemon, just as a run-and-remove container.

On NAS no. 1, I have used acme.sh to issue and deploy a wildcard certificate, that I would also like to deploy on Synology NAS no. 2 and also on another machine no. 3 using ssh.

I can deploy to NAS no. 1 from no. 1, no problem. I use `--deploy-hook synology_dsm`.

I can also deploy to NAS no. 2 from no. 1 using `--deploy-hook synology_dsm`.

And finally, I can deploy to machine no. 3 from NAS no. 1 using `--deploy-hook ssh`.

However, I can't figure out how to run the container using `--cron` to renew the certificate and automatically redeploy to all three locations. Is there a way to do this in one command? I have tried to use `--reloadcmd` with the first `--deploy` but that does not seem to work (the reload command script doesn't run).

Or do I have to use `--no-cron` initially, then later `--renew` via a scheduled task, and then do three separate manual deployments (assuming the renewal is successful)?

Thanks!

**Comment by rioncm** [comment]
check out this blog post and consider running the script directly on the NAS. It simplify's everything greatly. You can write a small shell script to schedule the renewals that renews the cert then deploy's it to your systems. 

[Synology DSM 7 with Let’s Encrypt and DNS Challenge](https://dr-b.io/post/Synology-DSM-7-with-Lets-Encrypt-and-DNS-Challenge)

**Comment by neilpang** [comment, marked as answer]
Multiple deploy hooks per cert are supported -- pass the flag repeatedly, they are saved together and every renewal redeploys all of them automatically:

```
acme.sh --deploy -d example.com --deploy-hook synology_dsm --deploy-hook ssh
```

What is not supported is the same hook twice with different credentials (two `synology_dsm` targets): the deploy config is saved per cert, so the second save overwrites the first and renewals only redeploy the last one. For NAS no. 2, run acme.sh there as well (issue + deploy locally), or push the cert over ssh as in the blog post above.
